src/coreclr/vm/riscv64/asmhelpers.S

```diff
@@ -894,43 +894,36 @@ LEAF_ENTRY JIT_ProfilerEnterLeaveTailcallStub, _TEXT
     ret
 LEAF_END JIT_ProfilerEnterLeaveTailcallStub, _TEXT
 
-// ------------------------------------------------------------------
-#define PROFILE_ENTER    1
-#define PROFILE_LEAVE    2
-#define PROFILE_TAILCALL 4
-#define SIZEOF__PROFILE_PLATFORM_SPECIFIC_DATA 272
-
 // ------------------------------------------------------------------
 .macro GenerateProfileHelper helper, flags
 NESTED_ENTRY \helper\()Naked, _TEXT, NoHandler
     // On entry:
-    //   t1 = functionIDOrClientID
-    //   t2 = profiledSp
+    //   t0 = functionIDOrClientID
+    //   t1 = profiledSp
     //   t6 = throwable
```

Comment:
Can't find who set arguments. You've changed register, should it be updated?

Reply:
The arguments are set in `CodeGen::genProfiling(Enter|Leave)Callback` in `codegenriscv64.cpp`. Yes, they should and are updated.

The reason I changed the registers is because t2 is used on RISC-V to store the call address of the stub by `genEmitHelperCall`.